### ISSUE TITLE
Ensure outputs meet Web Content Accessibility Guidelines (WCAG) 2.2 AA or better

### ISSUE THREAD

**Issue by eford** [fonsp/Pluto.jl#3435]
I'm hoping that someone can do an accessibility review, identify any things that need to be updated, and add a section to the documentation about any features that need to be avoided or patterns that need to be followed in order to ensure that Pluto outputs meet [Web Content Accessibility Guidelines (WCAG) 2.2](https://www.w3.org/TR/WCAG22/) at level AA or better.  

My understanding is that all large US universities will be required to have all their curricular materials meet this standard by April 24, 2026. So, in addition to the general desire to make outputs accessible, there is urgency to checking Pluto outputs against all the guidelines, so faculty are allowed to continue using Pluto in university settings.

**Comment by fonsp** [comment]
(You mean Pluto instead of Franklin, right?)

Pluto's interface is designed with the WCAG standards in mind, see https://github.com/fonsp/Pluto.jl/issues/3210 for a previous discussion. I would be very interested in a review!

Pluto cannot guarantee that the material you design meets these standards (e.g. you can embed an image in markdown without alt text). But it's a good idea to document _how_ to write accessible content in Pluto!

Could someone from a US institution find out exactly what the requirement is, and what we need to do? How is this done in other (web based) IDEs like rstudio, jupyter, replit, etc?

**Comment by eford** [comment]
Yes, I meant Pluto.  It's great that you already had WCAG standards in mind.  I was worried that it would be a humongous lift.  It sounds like there will probably be some effort required, but it won't be impossible.  
Of course, users will always have to use tools responsibly.  But I'd really appreciate a tutorial for how to write accessible notebooks and examples of common patterns (e.g., I don't know how to add an alt tag for a plot).
We've been given information about what to do for web, Word, and PDF files, but no guidance on what to do for IDEs.  

I'm not accessibility expert, but there are numerous webpage checker tools (see https://www.w3.org/WAI/test-evaluate/tools/list/ ) that can help identify issues.  I tried the bookmarklet at https://www.accessibilitycloud.com/lite/  on some simple demo notebooks and found several common issues.  

Would it be more helpful for me to post specifics in one big issue or to create separate issues for each error message?

**Comment by fonsp** [comment]
Hey! Posting the messages from automatic accessibility checkers is not very helpful (the Pluto devs can also run the tools themselves). But I would really appreciate feedback from:
- People who use accessibility tools
- People who understand the US requirements

A tutorial for writing accessible material is a good idea! https://github.com/JuliaPluto/website/issues/73

**Comment by eford** [comment]
Is there an easy way to override the css for colors of things that are intentionally light grey (e.g., runtime)?

**Comment by eford** [comment]
I'm getting errors about "Buttons do not have an accessible name" for the little checkmarks and upgrade symbols next to `using PkgName`

**Comment by fonsp** [comment]
> I'm getting errors about "Buttons do not have an accessible name" for the little checkmarks and upgrade symbols next to `using PkgName`

Fixed in da4430e1

**Comment by fonsp** [comment]
@eford You mentioned that a deadline is approaching. I think we still need to know exactly what the requirement is and how it applies to the Pluto project. Can you figure that out (or find someone who can) and write a summary of what we need to do?

**Comment by eford** [comment]
Here's the closest thing I've found to something official.  https://www.ada.gov/resources/2024-03-08-web-rule/

**Comment by eford** [comment]
I sent another email to my university folks asking for guidance.  When I asked late last year, they were overwhelmed mostly thinking about HTML, Word and PDF files.  Hopefully, by now some faculty member in another department has raised software development need and they'll have a more useful answer than when I first asked.
